edgexfoundry/edgex-go
# feat: Implement Device System Events

Also, refactored to use common Messaging Bootstrap Handler

closes #4099

Implements [related ADR](https://docs.edgexfoundry.org/2.3/design/adr/0024-system-events) 

Dependent on the following PRs:
- https://github.com/edgexfoundry/go-mod-core-contracts/pull/752
- https://github.com/edgexfoundry/go-mod-bootstrap/pull/360

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  link TBD

## Testing Instructions
Run edgex stack with Device Virtual
Stop Core Metadata container
Build and run this branch locally (require branches from PR referenced above) with DEBUG enabled
Using curl or postman delete, add and remove Device Virtual devices (can't use EdgeX UI from docker since core-metadata is running locally)
Verify log show debug message that System Event was published


## New Dependency Instructions (If applicable)
